arrow/ipc/file_reader.go

```diff
@@ -33,16 +33,166 @@ import (
 	"github.com/apache/arrow-go/v18/arrow/memory"
 )
 
-// FileReader is an Arrow file reader.
-type FileReader struct {
+type readerImpl interface {
+	getFooterEnd() (int64, error)
+	readFooter(*footerBlock) error
+	dict(memory.Allocator, *footerBlock, int) (dataBlock, error)
+	block(memory.Allocator, *footerBlock, int) (dataBlock, error)
+}
+
+type footerBlock struct {
+	offset int64
+	buffer *memory.Buffer
+	data   *flatbuf.Footer
+}
+
+type dataBlock interface {
+	Offset() int64
+	Meta() int32
+	Body() int64
+	NewMessage() (*Message, error)
+}
+
+type basicReaderImpl struct {
 	r ReadAtSeeker
+}
 
-	footer struct {
-		offset int64
-		buffer *memory.Buffer
-		data   *flatbuf.Footer
+func (r *basicReaderImpl) getFooterEnd() (int64, error) {
+	return r.r.Seek(0, io.SeekEnd)
+}
+
+func (r *basicReaderImpl) readFooter(f *footerBlock) error {
+	var err error
+
+	if f.offset <= int64(len(Magic)*2+4) {
```

Comment:
Can we add a variable for `len(Magic)*2+4` because we use this minimum size multiple times in this file?